src/record/api.rs

```diff
@@ -55,7 +56,113 @@ pub enum Row {
   Map(Vec<(Row, Row)>) // List of key-value pairs
 }
 
+// Macro to generate type-safe get_xxx methods e.g. get_bool, get_short
+macro_rules! row_primitive_accessor {
+  ($METHOD:ident, $VARIANT:ident, $TY:ty) => {
+    pub fn $METHOD(&self) -> Result<$TY, ParquetError> {
+        match *self {
+            Row::$VARIANT(v) => Ok(v),
+            _ => Err(ParquetError::General(format!("Cannot access {} as {}", self.get_type_name(), stringify!($VARIANT))))
```

Comment:
nit: maybe we can use `general_err!` here - a little less verbose
also we limit line width to 90 characters.